Test failure in CI build 7747

The following test appears to have failed:

[#7747](https://circleci.com/gh/cockroachdb/cockroach/7747):

```

W0921 17:36:04.204198 916 storage/store.go:1264  conflicting intents on ["\x00\x00\x00ka\x00\x01rdsc"]: resolved? true
W0921 17:36:04.211551 916 storage/store.go:1264  conflicting intents on ["\x00\x00\x00ka\x00\x01rdsc"]: resolved? true
I0921 17:36:04.213246 916 storage/replica_command.go:398  auto-gc'ed fcdb8204
W0921 17:36:04.223884 916 storage/replica.go:1465  batch resolve failed: at 0: key range "\x00\x00\x00ka\x00\x01rdsc"-"" outside of bounds of range ""-"a"
panic: intent resolution failed, error: at 0: key range "\x00\x00\x00ka\x00\x01rdsc"-"" outside of bounds of range ""-"a"

goroutine 6760 [running]:
github.com/cockroachdb/cockroach/storage.(*Replica).resolveIntents.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1469 +0x681
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc82039b8c0, 0xc82048a1e0, 0xc82031e800, 0x17)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:129 +0x2b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:131 +0x271

goroutine 1 [chan receive]:
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1217 +0xc39
github.com/cockroachdb/cockroach/storage_test.TestStoreRangeSplitConcurrent.func1(0xc82007cf10, 0x1, 0x8, 0xc82043a000, 0xc82007cf30, 0xc82007cf20)
	/go/src/github.com/cockroachdb/cockroach/storage/client_split_test.go:155 +0x2a2
created by github.com/cockroachdb/cockroach/storage_test.TestStoreRangeSplitConcurrent
	/go/src/github.com/cockroachdb/cockroach/storage/client_split_test.go:160 +0x264
FAIL	github.com/cockroachdb/cockroach/storage	24.566s
=== RUN   TestBatchBasics
I0921 17:35:47.820782 948 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0921 17:35:47.824790 948 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I0921 17:35:47.830481 948 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica.go
+++ b/storage/replica.go
@@ -1464,7 +1464,7 @@ func (r *Replica) resolveIntents(ctx context.Context, intents []proto.Intent) {
 				if log.V(1) {
 					log.Warningc(ctx, "batch resolve failed: %s", err)
 				}
-				if _, ok := err.(*proto.RangeKeyMismatchError); !ok {
+				if _, ok := unwrapIndexedError(err).(*proto.RangeKeyMismatchError); !ok {
 					// TODO(tschottdorf)
 					panic(fmt.Sprintf("intent resolution failed, error: %s", err.Error()))
 				}

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -65,13 +65,6 @@ func testRangeDescriptor() *proto.RangeDescriptor {
 	}
 }
 
-func unwrapIndexedError(err error) error {
-	if iErr, ok := err.(*errWithIndex); ok {
-		return iErr.err
-	}
-	return err
-}
-
 // boostrapMode controls how the first range is created in testContext.
 type bootstrapMode int
 

--- a/storage/store.go
+++ b/storage/store.go
@@ -148,6 +148,15 @@ func (ewi *errWithIndex) Error() string {
 	return fmt.Sprintf("at %d: %s", ewi.index, ewi.err)
 }
 
+// unwrapIndexedError returns the wrapped error for an *errWithIndex, and
+// the given error otherwise.
+func unwrapIndexedError(err error) error {
+	if iErr, ok := err.(*errWithIndex); ok {
+		return iErr.err
+	}
+	return err
+}
+
 type rangeAlreadyExists struct {
 	rng *Replica
 }

